pkg/util/ec2/imds_helpers.go

```diff
@@ -24,6 +24,15 @@ var (
 	imdsNetworkMacs = "/network/interfaces/macs"
 )
 
+// IMDSv2Usage is an enum to determine how to interact with the IMDSv2 option
+type IMDSv2Usage int
+
+const (
+	notUseIMDSv2 IMDSv2Usage = iota
+	useIMDSv2
+	forceIMDSv2
```

Comment:
What the difference between `useIMDSv2` and `forceIMDSv2` ? When would we have `useIMDSv2` not use IMDV2 ?

Is `forceIMDSv2` for when we only want to use IMDSv2 ? If so I'd rename this to (or something similar):
- `useIMDSv1`
- `useIMDSv2Andv1`
- `useIMDSv2`

Reply:
`forceIMDSv2` enforce the usage of v2 and if it fails it should not fallback on v1.
However we can rename them as you described

Reply:
done